### PR TITLE
[fix] OS-independent fix for the path parsing issue

### PR DESCRIPTION
**Pull request by the-non-feline**
OS-independent fix for #41 using `os.path`, **which also implements #42**

Tested and confirmed to work for Tutorials 1 and 2 and Examples 1 and 2 on Windows and Ubuntu. 

**Still need someone else to test on macOS as I do not own an Apple computer.**

### CODE REVIEW

**Comment by kevinkhu** [comment]
I tested this PR in a virtual environment with a new install of Bioverse with this PR. When running Tutorial1.ipynb, I encountered these warnings:
```
/home/kevin/.local/lib/python3.10/site-packages/pandas/core/computation/expressions.py:21: UserWarning: Pandas requires version '2.8.4' or newer of 'numexpr' (version '2.8.1' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED
/usr/lib/python3/dist-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.17.3 and <1.25.0 is required for this version of SciPy (detected version 1.26.3
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
  ```
  I ran ```pip install -U numexpr scipy``` and Tutorials 1 & 2, and Examples 1 & 2 ran without issue.

File: bioverse/util.py
Comment (matiscke) [inline bioverse/util.py:1]: good solution using os.path!

**Review by matiscke — APPROVED** [review]
Reviewed all files in this PR. This should fix the os-dependent issue with file parsing. Approved!

**Comment by matiscke** [comment]
@the-non-feline Did you run any tutorial/code from macOS that calls import_function_from_file ? If so, this is sufficiently tested and can be merged. Thanks!

**Pull request merged**